kyo-core/shared/src/main/scala/kyo/System.scala

```diff
@@ -0,0 +1,140 @@
+package kyo
+
+import System.Parser
+import java.lang.System as JSystem
+import kyo.kernel.Reducible
+
+abstract class System:
+    def env[E, A](name: String)(using Parser[E, A], Frame): Maybe[A] < (Abort[E] & IO)
+    def property[E, A](name: String)(using Parser[E, A], Frame): Maybe[A] < (Abort[E] & IO)
+    def lineSeparator(using Frame): String < IO
+    def userName(using Frame): String < IO
+    def userHome(using Frame): String < IO
+
+end System
+
+object System:
+
+    val local = Local.init(live)
+
+    val live: System =
+        new System:
+            def env[E, A](name: String)(using p: Parser[E, A], frame: Frame): Maybe[A] < (Abort[E] & IO) =
+                val value = JSystem.getenv(name)
+                if value == null then Maybe.empty
+                else p(value).map(Maybe(_))
```

Comment:
Can't this just be `Maybe(value).map(p)`?

Reply:
No, `Maybe.map` doesn't support effects so it produces a `kyo.Maybe[A < kyo.Abort[E]]`.